driver/device.c

```diff
@@ -122,6 +122,53 @@ DeviceIndicateConnectionStatus(NDIS_HANDLE MiniportAdapterHandle, NDIS_MEDIA_CON
     NdisMIndicateStatusEx(MiniportAdapterHandle, &Indication);
 }
 
+#pragma warning(disable : 4244)
+static VOID
+MagicConnectorIPV4NAT(_Inout_ WG_PEER *Peer, _Inout_ IPV4HDR *Hdr, _Inout_ UINT16 *NewVal, _Inout_ UINT16 *OldVal)
+{
+    /*
+     * MagicConnectorIPV4NAT redoes the NAT magic which we apply to connector
+     * RFC 1918 networks. Reverse of DemagicConnectorIPV4NAT.
+     * 
+     * If NewVal is zero, that would mean no modification was done. Callers may
+     * use this as an indication that checksum computation may be skipped.
+     */
+    UINT32_LE Saddr4 = Ntohl(Hdr->Saddr);
+    UINT8 FirstOctet = (Saddr4 >> 24) & 0xFFFFFFFF;
+    UINT32 FirstOctetBitMask = 0x00FFFFFF;
+
+    *OldVal = (UINT16)(Hdr->Saddr & 0x0000FFFF);
+
+    KIRQL Irql = ExAcquireSpinLockShared(&Peer->EndpointLock);
+    UINT8 CurrentENatIndex = Peer->ENatIndex, CurrentONatIndex = Peer->ONatIndex;
+    ExReleaseSpinLockShared(&Peer->EndpointLock, Irql);
+
+    // TODO: Need lookup for this AT.
+    if (FirstOctet == 10)
+    {
+        Saddr4 = (Saddr4 & FirstOctetBitMask) | (CurrentENatIndex << 24);
+        Hdr->Saddr = Htonl(Saddr4);
+        *NewVal = (UINT16)(Hdr->Saddr & 0x0000FFFF);
+        return;
+    }
+    else if (FirstOctet == 172)
```

Comment:
I think this could be `if` instead of `else if`

Reply:
Done.